indico/util/user.py

```diff
@@ -170,5 +192,25 @@ def principal_from_identifier(identifier, allow_groups=False, allow_external_use
         if event_role is None or event_role.event_id != event_id:
             raise ValueError('Invalid event role: {}'.format(event_role_id))
         return event_role
+    elif type_ == 'CategoryRole':
+        if not allow_category_roles:
+            raise ValueError('Category roles are not allowed')
+
+        event = Event.get(event_id)
+        if event is None:
+            raise ValueError('Invalid event id: {}'.format(event_id))
+        try:
+            category_role_id = int(data)
+        except ValueError:
+            raise ValueError('Invalid data')
+        available_category_roles = list(
+            itertools.chain.from_iterable(
```

Comment:
see my above comment ;)

FWIW, for the case here the staticmethod would need another argument (false by default) that skips the subquery filter in case of `soft_fail`. Or you can just use `CategoryRole.get()` in case of `soft_fail`, and otherwise the staticmethod that checks the ancestry.